### PR TITLE
CSS test fonts

### PR DESCRIPTION
**Pull request by gsnedders**
This adds an up-to-date set of CSS tests fonts, specifically the subset that can be installed for all tests.

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/8374)

### CODE REVIEW

**Comment by eefriedman** [comment]
```
Running 6848 tests in web-platform-tests

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-family-applies-to-001.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-family-applies-to-002.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-family-applies-to-005.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-family-applies-to-007.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-family-applies-to-008.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-family-applies-to-009.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-family-applies-to-010.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-family-applies-to-011.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-family-applies-to-014.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-family-applies-to-017.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-family-applies-to-015.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-001.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-002.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-004.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-005.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-012.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-013.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-015.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-016.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-023.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-024.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-026.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-027.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-034.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-035.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-037.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-045.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-038.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-046.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-048.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-049.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-056.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-057.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-059.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-067.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-060.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-070.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-068.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-071.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-078.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-079.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-082.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-089.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-081.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-090.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-092.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-100.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-093.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-101.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-102.htm

  ▶ PASS [expected FAIL] /css-fonts-3_dev/html/font-size-122.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/block-in-inline-remove-000.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/block-in-inline-remove-002.htm

  ▶ FAIL [expected PASS] /css21_dev/html4/c62-percent-000.htm
  └   → /css21_dev/html4/c62-percent-000.htm 5d2ab33ac55934ac77c2d2598ec5ff38b5715be1
/css21_dev/html4/reference/c62-percent-000-ref.htm f1fcf1064b8a73db5de471e0f9b921cfb84b9dc5
Testing 5d2ab33ac55934ac77c2d2598ec5ff38b5715be1 == f1fcf1064b8a73db5de471e0f9b921cfb84b9dc5

  ▶ PASS [expected FAIL] /css21_dev/html4/floats-wrap-bfc-002-right-table.htm

Ran 6848 tests finished in 1093.0 seconds.
  • 6793 ran as expected. 10 tests skipped.
  • 1 tests failed unexpectedly
  • 54 tests passed unexpectedly
```

**Comment by jdm** [comment]
The only failure is #7791.

**Comment by gsnedders** [comment]
Hmm. All of /css-fonts-3_dev/html/font-size-* reproducibly fail here…Will look into it. Probably more just intermittent failures.

**Comment by gsnedders** [comment]
These seem to all pass reliably on OS X, and fail on Linux.

**Comment by gsnedders** [comment]
Can anyone give any insight into that failure? Or what I should do?

**Comment by Manishearth** [comment]
It's an intermittent, don't worry.

**Comment by jdm** [comment]
I don't see the point of retrying on try for known intermittents.

**Comment by larsbergstrom** [comment]
hrm, I may have to switch the android builder to something with more RAM

**Comment by gsnedders** [comment]
So as far as I'm concerned, all tests are known intermittents.

File: resources/css-test-fonts.css [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+@font-face {
+    font-family: "White Space";
+    src: url(css-test-fonts/ahem-extra/AHEM_WhiteSpace.TTF);
+}
+
+/*
+
+These can't be globally enabled, because of tests checking that
+'sans-serif', Ahem falls back to Ahem, rather than using the generic
+sans-serif.
```
Comment (SimonSapin) [inline resources/css-test-fonts.css]: It should be possible to enable them once https://github.com/servo/servo/issues/8371 is fixed, right?

File: resources/css-test-fonts/LICENSE [hunk 2]
```diff
@@ -0,0 +1,3 @@
+This repository is covered by the dual-licensing approach described in:
```
Comment (SimonSapin) [inline resources/css-test-fonts/LICENSE]: Replace repository with directory here?

**Comment by eefriedman** [comment]
```
  ▶ CRASH [expected OK] /dom/nodes/Document-createElement-namespace.html
  │ 
  │ thread 'FontCacheTask' panicked at 'assertion failed: libc::close(self.fd) == 0', /home/servo/.cargo/git/checkouts/ipc-channel-d95a23d1f1577bfc/master/platform/linux/mod.rs:101
  │ stack backtrace:
  │ thread 'file_loader' panicked at 'called `Result::unwrap()` on an `Err` value: ()', src/libcore/result.rs:736
  │ thread 'ResourceManager' panicked at 'index out of bounds: the len is 0 but the index is 0', src/libcollections/vec.rs:1044
  │    1:     0x7f337ac769d0 - sys::backtrace::tracing::imp::write::h23bfa7a49ed99d87Klt
  │    2:     0x7f337ac78f15 - panicking::log_panic::_<closure>::closure.39180
  │    3:     0x7f337ac78961 - panicking::log_panic::h846e3da4ec5bd0b1Wlx
  │    4:     0x7f337ac65b43 - sys_common::unwind::begin_unwind_inner::h04ad996baf6f81c7bds
  │    5:     0x7f337a39ac7f - sys_common::unwind::begin_unwind::begin_unwind::h17276457297030504697
  │    6:     0x7f337a3941e4 - collections..vec..Vec<platform..linux..UnixChannel>::drop.14769::hf42e993ae2f41315
  │    7:     0x7f337a39b8d1 - platform::linux::_<impl>::send::h833228e79649204dKMa
  │    8:     0x7f337a17bc6a - ipc::_<impl>::send::send::h11444091711227529263
  │    9:     0x7f337a17a085 - _<impl>::load_async::h08b3efe118c0f69a9kh
  │   10:     0x7f3379f51848 - font_cache_task::_<impl>::run::h9249a0f8ae6347c99tk
  │   11:     0x7f3379f70d65 - boxed::_<impl>::call_box::call_box::h11716247010833971494
  │   12:     0x7f3379f70004 - sys_common::unwind::try::try_fn::try_fn::h15530385003288815698
  │   13:     0x7f337ac75f08 - __rust_try
  │   14:     0x7f337ac724db - sys_common::unwind::try::inner_try::h25c781202f1ad3a1J9r
  │   15:     0x7f3379f70194 - boxed::_<impl>::call_box::call_box::h9278030496435425020
  │   16:     0x7f337ac77a03 - sys::thread::_<impl>::new::thread_start::h89508aa1f401717d8Dw
  │   17:     0x7f3376ce0181 - start_thread
  │   18:     0x7f33774fa47c - __clone
  │   19:                0x0 - <unknown>
```

This looks bad... but I'm not sure what could cause it.  Maybe @pcwalton has an idea?

**Comment by jdm** [comment]
Looks like #8082.

**Comment by jdm** [comment]
@bors-servo: retry
* #8082

**Comment by jdm** [comment]
It's happened twice and it's happening in the font cache task, which seems particularly relevant here.

**Comment by larsbergstrom** [comment]
```
CRASH [expected OK] /dom/nodes/Document-createElement-namespace.html
  │ 
  │ thread 'FontCacheTask' panicked at 'assertion failed: libc::close(self.fd) == 0', /home/servo/.cargo/git/checkouts/ipc-channel-d95a23d1f1577bfc/master/platform/linux/mod.rs:101
  │ stack backtrace:
  │ thread 'file_loader' panicked at 'called `Result::unwrap()` on an `Err` value: ()', src/libcore/result.rs:736
  │ thread 'ResourceManager' panicked at 'assertion failed: libc::close(self.fd) == 0', /home/servo/.cargo/git/checkouts/ipc-channel-d95a23d1f1577bfc/master/platform/linux/mod.rs:101
  │ thread 'file_loader' panicked at 'called `Result::unwrap()` on an `Err` value: ()', src/libcore/result.rs:736
  │ thread 'file_loader' panicked at 'called `Result::unwrap()` on an `Err` value: ()', src/libcore/result.rs:736
  │    1:     0x557e4dcb9410 - sys::backtrace::tracing::imp::write::h23bfa7a49ed99d87Klt
  │    2:     0x557e4dcbb955 - panicking::log_panic::_<closure>::closure.39180
  │    3:     0x557e4dcbb3a1 - panicking::log_panic::h846e3da4ec5bd0b1Wlx
  │    4:     0x557e4dca8583 - sys_common::unwind::begin_unwind_inner::h04ad996baf6f81c7bds
  │    5:     0x557e4d3dbc6f - sys_common::unwind::begin_unwind::begin_unwind::h13278613625421187389
  │    6:     0x557e4d3d51d4 - collections..vec..Vec<platform..linux..UnixChannel>::drop.14769::hf42e993ae2f41315
  │    7:     0x557e4d3dc8c1 - platform::linux::_<impl>::send::h833228e79649204dKMa
  │    8:     0x557e4d1bc73a - ipc::_<impl>::send::send::h14078470962244269638
  │    9:     0x557e4d1bab55 - _<impl>::load_async::h08b3efe118c0f69a9kh
  │   10:     0x557e4cf92318 - font_cache_task::_<impl>::run::h9249a0f8ae6347c99tk
  │   11:     0x557e4cfb1835 - boxed::_<impl>::call_box::call_box::h6863437181062840643
  │   12:     0x557e4cfb0ad4 - sys_common::unwind::try::try_fn::try_fn::h1664613225726501668
  │   13:     0x557e4dcb8948 - __rust_try
  │   14:     0x557e4dcb4f1b - sys_common::unwind::try::inner_try::h25c781202f1ad3a1J9r
  │   15:     0x557e4cfb0c64 - boxed::_<impl>::call_box::call_box::h11915057183127424801
  │   16:     0x557e4dcba443 - sys::thread::_<impl>::new::thread_start::h89508aa1f401717d8Dw
  │   17:     0x7fba886db181 - start_thread
  │   18:     0x7fba88ef547c - __clone
  │   19:                0x0 - <unknown>
```

Hrm, this looks like the linux version of https://github.com/pcwalton/ipc-channel/pull/10, which I *thought* we had done, but can't find. I'll try to look into this Monday.

File: resources/css-test-fonts.css (hunk 1 above)
Comment (gsnedders) [inline resources/css-test-fonts.css]: No. The CSS testsuite has tests that 'sans-serif' isn't recognised as the generic and falls back to the next thing in the list, and has tests that 'sans-serif' matches an installed font with that name.

**Comment by gsnedders** [comment]
This should still have the S-tests-failed label, but not S-needs-review. Someone should fix that. And @highfive.

**Comment by jdm** [comment]
Highfive clears existing labels and applies the awaiting review one one new pushes, since the default assumption is that new changes require review and old results are not meaningful.

**Comment by Wafflespeanut** [comment]
▶ TIMEOUT [expected OK] /_mozilla/mozilla/mime_sniffing_font_context.html

    ▶ Unexpected subtest result in /_mozilla/mozilla/mime_sniffing_font_context.html:
    └ NOTRUN [expected PASS] Sniffed font is loaded and applied

    ▶ Unexpected subtest result in /_mozilla/mozilla/mime_sniffing_font_context.html:
    │ FAIL [expected PASS] XHR Content-Type has xhtml+xml
    │   → assert_equals: expected "application/xhtml+xml" but got "application/json"
    │ 
    │ xhr.onload<@http://web-platform.test:8000/_mozilla/mozilla/mime_sniffing_font_context.html:20:5
    │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1381:20
    └ Test.prototype.step_func_done/<@http://web-platform.test:8000/resources/testharness.js:1421:1

    ▶ Unexpected subtest result in /_mozilla/mozilla/mime_sniffing_font_context.html:
    │ FAIL [expected PASS] XHR Content-Type has xml
    │   → assert_equals: expected "application/xml" but got "application/json"
    │ 
    │ xhr.onload<@http://web-platform.test:8000/_mozilla/mozilla/mime_sniffing_font_context.html:30:5
    │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1381:20
    └ Test.prototype.step_func_done/<@http://web-platform.test:8000/resources/testharness.js:1421:1

    ▶ Unexpected subtest result in /_mozilla/mozilla/mime_sniffing_font_context.html:
    │ FAIL [expected PASS] XHR Content-Type is empty
    │   → assert_equals: expected (object) null but got (string) "application/json"
    │ 
    │ xhr.onload<@http://web-platform.test:8000/_mozilla/mozilla/mime_sniffing_font_context.html:9:1
    │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1381:20
    └ Test.prototype.step_func_done/<@http://web-platform.test:8000/resources/testharness.js:1421:1

**Comment by KiChjang** [comment]
@delegate+

I'm just gonna give you some privileges here; hopefully it'll help you in working on this PR.

**Comment by KiChjang** [comment]
@bors-servo delegate+

I messed up big time :)

**Comment by gsnedders** [comment]
Looks like the font cache panic is gone, at least.

**Comment by gsnedders** [comment]
Bah, back to the font cache failures :(

**Comment by jdm** [comment]
@gsnedders Are you still working on this? I'm just triaging PRs and don't see any activity for the past month.

**Comment by gsnedders** [comment]
@jdm Not actively. With the font cache panic still happening occasionally (despite my previous comment), we can't land it. So it's blocked on fixing that.

**Comment by jdm** [comment]
@bors-servo: try
We need a new log for #9770.

**Comment by jdm** [comment]
Linux output is https://gist.github.com/jdm/551a92fb9a5d68b2e5cf .

**Comment by jdm** [comment]
I think the best course of action here is to log into a builder and try to catch one of the failures in a debugger (or rr?).

**Comment by jdm** [comment]
Huh, these are all tests that contain a bunch of iframes. I guess we're creating a bunch of new threads with each font loaded, and that gets multiplied per iframe...

**Comment by jdm** [comment]
I propose seeing what happens if https://github.com/gsnedders/servo/blob/03c7bc267bdcad1f073a5600433e435db22f8ce0/components/layout/layout_thread.rs#L341 is moved outside of the loop so it can be reused. My understanding is that IPC channels consume file descriptors, so this would be a significant reduction given how many fonts are requested by the CSS file.

**Comment by nox** [comment]
Superseded by #11311.

**Pull request closed without merging**